Modules/Course/classes/class.ilObjCourseAccess.php

```diff
@@ -75,6 +78,13 @@ function _checkAccess($a_cmd, $a_permission, $a_ref_id, $a_obj_id, $a_user_id =
 		{
 			$a_user_id = $ilUser->getId();
 		}
+
+		// cat-tms-patch start
+		$decision = self::_checkAccessExtension($a_cmd, $a_permission, $a_ref_id, $a_obj_id, $a_user_id);
```

Comment:
Kommt der Benutzer hier immer durch auch wenn er nur in die Settings will? Wenn ja wäre es nicht richtig, dass hier nur dann zu machen, wenn er auch wirklich versucht den Kurs zu kopieren.

Reply:
Ja, Hier geht es bei jedem Check vorbei. Ich wollte die Änderung hier minimal halten um möglichst viel Code in "unseren" Bereich zu ziehen und da testbar zu machen und würde dafür auch zusätzliche Calls in Kauf nehmen.

Reply:
Dann muss der check on copy permission aber noch in das ToDo unten von der Funktion. Sonst geht das unter er prüft immer ob er kopieren darf.

Reply:
Hab ich [drin](https://github.com/conceptsandtraining/ILIAS/pull/1331/files#diff-055d05b3de3ba56c1f35ac6861b36e20R15) oder ich verstehs nicht.

Reply:
Ach verdammt. Hab das if überlesen...